src/test/test_pidfile.sh

```diff
@@ -71,7 +71,7 @@ function TEST_pidfile() {
     # another daemon
     cp $dir/osd.0.pid $dir/osd.0.pid.old  # so that kill_daemon finds the pid
```

Comment:
We need kill_daemons to kill the daemons (reason why it must find the old pid) and observe that it does not remove $dir/osd.0.pid (because it either is a different path/inode or because its content has changed).